sematic/types/types/aws/s3.py

```diff
@@ -0,0 +1,22 @@
+# Standard Library
+from dataclasses import dataclass
+
+
+@dataclass
+class S3Bucket:
+    """
+    Basic type to describe an S3 bucket.
+    """
+
+    region: str
+    name: str
+
+
+@dataclass
+class S3Location:
```

Comment:
I think this will be a lot more usable if you have a classmethod that can be used like:

```
# region is optional
S3Location.from_uri("s3://foo/bar/baz", region=region)
```

Reply:
Done.